### DISCUSSION TITLE
Camomile's LV2 on Raspberry Pi

### DISCUSSION THREAD

**Discussion by Gnuv** [pierreguillot/Camomile#225] [Q&A]
Hi Pierre,
Is Camomile’s LV2 run on Raspberry Pi ARM or just x86 computers ? 

I mostly think about it for deploying plugins on the MOD-DUO project (https://www.moddevices.com/) running in raspberry like Zynthian (https://wiki.zynthian.org/index.php/MOD-HOST_%26_MOD-UI) or MODEP (https://blokas.io/modep/).

There’s a discussion here : https://forum.moddevices.com/t/granular-and-sampler-progress-for-the-mod-duo-x/4503/3 and this note : « The Camomile integration is under testing phase » here : https://github.com/moddevices/mod-plugin-builder, but that’s all that I find for now.

Thanks

**Comment by pierreguillot** [comment]
Hi,

The pre-built binaries are for x86 but I don't see any reason to not be compatible with ARM. I'm not familiar with MOD-DUO but a first step would be to compile the plugin for ARM. I can't do that now (I have a raspberry pi but I don't have any monitor and keyboard...). But if you can, please try it. I suggest you use the same approach as the one used on [travis](https://github.com/pierreguillot/Camomile/blob/master/.travis.yml#L13l) for Linux. 

**Important**: 
- Clone the repository recursively as described in the README.md otherwise it won't compile
- In the script section, replace *TARGET_ARCH=-m64* with the configuration that you need

Tell me if you encounter any issue.

**Reply by Gnuv** [reply]
Thanks for your reply.
I will try this when I can. I’m not so confident in compiling for linux for now.
Note that you don’t need a keyboard and a monitor to use ZynthianOS with MOD-UI. I just write a tuto here in english : https://wiki.zynthian.org/index.php/Beginners_Headless_Zynthian_Tutorial and here in french (more updated) : https://wiki.zynthian.org/index.php/Tutoriel_Zynthian_Headless_pour_d%C3%A9butants_(french)

**Reply by pierreguillot** [reply]
I don't really have time to work on this now. But do you think it would be useful to add another target architecture for Linux ARM directly in the distribution? If so, I can try to do that with Travis CI. And I guess if I manage to use CMake, as requested [here](https://github.com/pierreguillot/Camomile/issues/212), it should not require too many changes.

**Reply by Gnuv** [reply]
I think so. OS like Zynthian with MOD-UI or MODEP in Raspberry Pi are really great and easy to use. This is an affordable way to have autonomous patch readers for live performances without the need of complex and expensive laptops. Our digital orchestra in Nice is now entirely powered by these single board computers. For musicians, that works just like an acoustic/electric instrument : power and play. For composers who program in pure data or other languages, there isn’t really a need for custom LV2, as these softwares runs natively on the raspberry. But for students and composers that aren’t programmers, having easy to use LV2 tools programmed in pd with Camomile inside MOD-UI and/or Zynthian UI could be a great thing. If you open this possibility, we will program some for advance midi mapping and cue list management for the digital music interfaces that we have.

**Reply by pierreguillot** [reply]
I did a quick investigation of the issue. So it seems that Juce has not been tested on a Raspberry Pi ARM (only Ubuntu) and Github action that I use now for the CI doesn't support the Linux arm architecture (extension can be used but it becomes even more complicated to debug). So I suggest, you try to compile following the README instruction and let me know if you encounter any issue.
